### PR TITLE
docs: Use consistent structure for Flow component references

### PR DESCRIPTION
**Pull request by rfratto**
This PR changes the component references to use a consistent structure:

* Examples have been moved to the bottom of the page, and a Usage section has been moved to where Examples used to be. 
* Components that have blocks now have a h2 for `Blocks` and one h3 per defined block.

### CODE REVIEW

**Comment by rfratto** [comment]
It's probably best to run `make docs` from docs/ to be able to review this locally to see how it displays.

File: docs/sources/flow/reference/components/prometheus.relabel.md [hunk 1]
```diff
@@ -29,23 +29,17 @@ each metric in order of their appearance in the configuration file.
...
-
-	rule {
-		action        = "keep"
-		source_labels = ["app"]
-		regex         = "backend"
-	}
+prometheus.relabel "LABEL" {
+  forward_to = RECEIVER_LIST
+
+  rule {
```
Comment (tpaschalis) [inline docs/sources/flow/reference/components/prometheus.relabel.md:38]: Let's sync this with the discovery.relabel and either have none or both of them displaying dummy rules here?

**Review by tpaschalis — APPROVED** [review]
LGTM ✉️

File: docs/sources/flow/reference/components/discovery.kubernetes.md [hunk 2]
```diff
@@ -388,3 +393,39 @@ values.
...
+```
+
+### Kubeconfig authentication
+
+This example uses a kubeconfig file to authenticate to the Kubernetes API:
+
+```river
+discovery.kubernetes "k8s_pods" {
+  role = "pod"
+  kubeconfig_file = "/path/to/kubeconfig"
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:416]: PATH_TO_KUBECONFIG or PATH?

File: docs/sources/flow/reference/components/local.file.md [hunk 3]
```diff
@@ -94,3 +94,11 @@ component.
   timestamp, in Unix seconds, that the file was last sucessfully accessed.
 
 [secret]: ../secrets.md#is_secret-argument-in-components
+
+## Example
+
+```river
+local.file "my_file" {
+  filename = "path/to/my/file"
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/local.file.md:102]: PATH?

File: docs/sources/flow/reference/components/discovery.kubernetes.md [hunk 4]
```diff
@@ -50,14 +36,6 @@ The `role` argument is required to specify what type of targets to discover.
...
 
-The following sub-blocks are supported:
-
-Name | Description | Required
----- | ----------- | --------
-[`namespaces`](#namespaces-block) | Information about which Kubernetes namespaces to search. | no
-[`selectors`](#selectors-block) | Selectors to limit objects selected. | no
-[`http_client_config`](#http_client_config-block) | HTTP client configuration for Kubernetes requests. | no
-
 ### `node` role
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:39]: Not explicitly related to the PR but should `node` become node to match the change in blocks below?

File: docs/sources/flow/reference/components/discovery.kubernetes.md (hunk 2 above)
Comment (rfratto) [reply docs/sources/flow/reference/components/discovery.kubernetes.md:416]: Ah, this is supposed to be an example so it's not a placeholder. I'll make up a fake path.

File: docs/sources/flow/reference/components/local.file.md (hunk 3 above)
Comment (rfratto) [reply docs/sources/flow/reference/components/local.file.md:102]: Ah, this is supposed to be an example so it's not a placeholder. I'll make up a fake path.

File: docs/sources/flow/reference/components/discovery.kubernetes.md (hunk 4 above)
Comment (mattdurham) [reply docs/sources/flow/reference/components/discovery.kubernetes.md:39]: Applies to all the other roles also

Comment (rfratto) [reply docs/sources/flow/reference/components/discovery.kubernetes.md:39]: Yep, just did that 👍

File: docs/sources/flow/reference/components/prometheus.relabel.md (hunk 1 above)
Comment (rfratto) [reply docs/sources/flow/reference/components/prometheus.relabel.md:38]: Synced them 👍

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 5]
```diff
@@ -354,3 +340,31 @@ information.
   remote storage.
 * `prometheus_remote_storage_exemplars_in_total` (counter): Exemplars read into
   remote storage.
+
+## Example
+
```
Comment (karengermond) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:345]: Later might want to add an introduction to each example to say what's going on. The comments basically do the trick but it's better to have something in the text.

**Review by karengermond — APPROVED** [review]
Love it!

**Pull request merged**